lxd/resource_lxd_container.go

```diff
@@ -344,12 +348,28 @@ func resourceLxdContainerRead(d *schema.ResourceData, meta interface{}) error {
 	}
 	log.Printf("[DEBUG] Retrieved container state %s:\n%#v", name, state)
 
+	d.Set("ephemeral", container.Ephemeral)
+	d.Set("privileged", false) // Create has no handling for it yet
+
+	config := make(map[string]string)
+	limits := make(map[string]string)
 	for k, v := range container.Config {
 		if strings.Contains(k, "limits.") {
-			log.Printf("[DEBUG] Setting limit %s: %s", k, v)
-			d.Set(k, v)
+			limits[strings.TrimPrefix(k, "limits.")] = v
+		} else if strings.HasPrefix(k, "boot.") {
+			config[k] = v
+		} else if strings.HasPrefix(k, "environment.") {
+			config[k] = v
+		} else if strings.HasPrefix(k, "raw.") {
+			config[k] = v
+		} else if strings.HasPrefix(k, "security.") {
+			config[k] = v
+		} else if strings.HasPrefix(k, "user.") {
+			config[k] = v
```

Comment:
Is it possible to use a `switch` statement here? It's usually cleaner to use a switch instead of multiple `else if` statements. If switch isn't possible, no problem.

Reply:
afaik i can't do a switch with the hasPrefix call

Reply:
Yes, you are absolutely correct.